lib/synapse/haproxy.rb

```diff
@@ -686,7 +686,7 @@ def generate_frontend_stanza(watcher, config)
       stanza = [
         "\nfrontend #{watcher.name}",
         config.map {|c| "\t#{c}"},
-        "\tbind #{@opts['bind_address'] || 'localhost'}:#{watcher.haproxy['port']}",
+        "\tbind #{ watcher.haproxy['bind_address'] || @opts['bind_address'] || 'localhost'}:#{watcher.haproxy['port']}",
```

Comment:
Let's use this as an opportunity to add a generate_frontend_stanza test to the haproxy_spec :-)